pkg/server/settings_cache.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package server
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+)
+
+// storeCachedSettingsKVs stores or caches node's settings locally.
+// This helps in restoring the node restart with the at least the same settings with which it died.
+func storeCachedSettingsKVs(ctx context.Context, eng storage.Engine, kvs []roachpb.KeyValue) error {
+	batch := eng.NewBatch()
+	defer batch.Close()
+	for _, kv := range kvs {
+		kv.Value.Timestamp = hlc.Timestamp{} // nb: Timestamp is not part of checksum
+		if err := storage.MVCCPut(
+			ctx, batch, nil, keys.StoreCachedSettingsKey(kv.Key), hlc.Timestamp{}, kv.Value, nil,
+		); err != nil {
+			return err
+		}
+	}
+	return batch.Commit(false /* sync */)
```

Comment:
Drive by comment: I'm curious why we're choosing to not durably persist these writes here. I think we want to, no?